node/bft-consensus/src/state.rs

```diff
@@ -130,8 +130,16 @@ impl<N: Network, C: ConsensusStorage<N>> ExecutionState for BftExecutionState<N,
                 return Ok(None);
             }
 
+            // Timestamps are on the batches, not the transactions. We assume every batch contains
+            // at least one valid transaction.
+            let mut timestamps =
+                consensus_output.batches.iter().map(|batch| batch.0.metadata.created_at).collect::<Vec<_>>();
```

Comment:
Actually, we need to compute the median timestamp from the certificates in the `sub_dag`.